server/src/main/java/com/thoughtworks/go/server/dashboard/GoDashboardCurrentStateLoader.java

```diff
@@ -76,8 +78,14 @@ public GoDashboardCurrentStateLoader(PipelineDao pipelineDao, TriggerMonitor tri
     }
 
     public List<GoDashboardPipeline> allPipelines(CruiseConfig config) {
-        List<String> pipelineNames = CaseInsensitiveString.toStringList(config.getAllPipelineNames());
-        PipelineInstanceModels historyForDashboard = loadHistoryForPipelines(pipelineNames);
+        List<CaseInsensitiveString> allPipelineNames = config.getAllPipelineNames();
+
+        HashSet<CaseInsensitiveString> newPipelineNames = new HashSet<>(allPipelineNames);
+        if (!newPipelineNames.equals(previousPipelineNames)) {
+            List<String> pipelineNames = CaseInsensitiveString.toStringList(allPipelineNames);
+            historyForDashboard = loadHistoryForPipelines(pipelineNames);
```

Comment:
Dont we need to update `historyForDashboard` on pipeline-config-api callback?

Reply:
which callback is this?